ray-operator/controllers/ray/rayservice_controller_test.go

```diff
@@ -348,10 +353,16 @@ var _ = Context("Inside the default namespace", func() {
 
 			// The cluster shouldn't switch until deployments are finished updating
 			fakeRayDashboardClient.SetServeStatus(generateServeStatus(metav1.NewTime(time.Now().Add(time.Duration(-5)*time.Minute)), "UPDATING"))
-			myRayService.Spec.RayClusterSpec.HeadGroupSpec.Template.Spec.Containers[0].Env[1].Value = "UPDATED_VALUE"
-			myRayService.Spec.RayClusterSpec.WorkerGroupSpecs[0].Template.Spec.Containers[0].Env[1].Value = "UPDATED_VALUE"
 
-			Expect(k8sClient.Update(ctx, myRayService)).Should(Succeed(), "failed to update test RayService resource")
+			err := retry.RetryOnConflict(retry.DefaultRetry, func() error {
+				Eventually(
```

Comment:
Can you explain why we need to `getResourceFunc` here?

Reply:
Without getResourceFunc , if a 409 conflict happens, the resourceVersion we pass to API server during retry updating will never be updated and thus will never match the resourceVersion in the API server.